pkgdb/include/flox/registry.hh

```diff
@@ -162,8 +162,8 @@ struct RegistryInput : public InputPreferences
   [[nodiscard]] bool
   operator==( const RegistryInput & other ) const
   {
-    if ( static_cast<InputPreferences>( *this )
-         != static_cast<InputPreferences>( other ) )
+    if ( *dynamic_cast<InputPreferences const *>( this )
+         != dynamic_cast<const InputPreferences &>( other ) )
```

Comment:
For my own knowledge why would we want to do this at runtime?

Reply:
The warning was related to 'slicing data', casting to the base object.  BUT I think I passed over it too quickly and this may still not address the issue?  I think I was adding some type checking by changing to `dynamic_cast` but I'm not 100% convinced that's the case.... Running the linter again this afternoon and I'll take another look at this.